tests/manage/z_cluster/cluster_expansion/test_vm_delete_and_replace.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.resources import pod
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import (
+    tier4a,
+    E2ETest,
+    vsphere_platform_required,
+    skipif_vsphere_ipi,
+    bugzilla,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.utility import vsphere
+from ocs_ci.ocs.node import verify_all_nodes_created, get_osd_running_nodes
+
+
+logger = logging.getLogger(__name__)
+
+
+@tier4a
+@skipif_vsphere_ipi
+@vsphere_platform_required
+@bugzilla("1904302")
+@pytest.mark.polarion_id("OCS-2658")
+class TestVmDeletionAndReplacement(E2ETest):
+    """
+    Test to delete VM with data disk and replacing with another node
+    """
+
+    @pytest.fixture(autouse=True)
+    def init(self):
+
+        self.cluster_name = config.ENV_DATA.get("cluster_name")
+        self.cluster = config.ENV_DATA["vsphere_cluster"]
+        self.datacenter = config.ENV_DATA["vsphere_datacenter"]
+        self.datastore = config.ENV_DATA["vsphere_datastore"]
+        self.server = config.ENV_DATA["vsphere_server"]
+        self.user = config.ENV_DATA["vsphere_user"]
+        self.password = config.ENV_DATA["vsphere_password"]
+        self.sanity_helpers = Sanity()
+
+    def test_vm_deletion_and_replacement(
+        self, pvc_factory, pod_factory, add_nodes, bucket_factory, rgw_bucket_factory
+    ):
+        """
+        1. Get OSDs and OSD running nodes
+        2. Get compute VMs
+        3. Destroy VM
+        4. Replace VM
+        5. Label node as OCS
+        6. Check all pods are running and cluster in Healthy state
+        7. Create Resources
+        """
+
+        # OSDs before running VM replacement
+        old_osds = []
+        for osd in pod.get_osd_pods():
+            old_osds.append(osd.name)
+        logger.info(f"OSDs before VM replacement {old_osds}")
+
+        # Get osd running nodes
+        osd_nodes = get_osd_running_nodes()
+        logger.info(f"osd_nodes {osd_nodes}")
+
+        # Get compute VMs from the cluster
+        vm_obj = vsphere.VSPHERE(self.server, self.user, self.password)
+        old_vms = vm_obj.get_compute_vms_in_pool(
+            name=self.cluster_name, dc=self.datacenter, cluster=self.cluster
+        )
+        logger.info(f"Type of vm {type(old_vms)}")
```

Comment:
If I understand it correctly this will write:
`Type of vm list`

For what is this information useful? I think you can remove it and just keep info log message below.

Reply:
forgot to remove after the local test, now removed